src/ZoneServer/Skills/Handlers/Scout/DaggerSlash.cs

```diff
@@ -0,0 +1,125 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using Melia.Shared.L10N;
+using Melia.Shared.Tos.Const;
+using Melia.Shared.World;
+using Melia.Zone.Network;
+using Melia.Zone.Skills.Combat;
+using Melia.Zone.Skills.Handlers.Base;
+using Melia.Zone.Skills.SplashAreas;
+using Melia.Zone.World.Actors;
+using Melia.Zone.World.Actors.CombatEntities.Components;
+using static Melia.Zone.Skills.SkillUseFunctions;
+using System.Threading.Tasks;
+
+namespace Melia.Zone.Skills.Handlers.Scout
+{
+	/// <summary>
+	/// Handler for the Scout skill Dagger Slash.
+	/// </summary>
+	[SkillHandler(SkillId.Scout_DaggerSlash)]
+	public class DaggerSlash : IGroundSkillHandler
+	{
+		/// <summary>
+		/// Handles skill, do a slash attack to the nearby enemies.
+		/// </summary>
+		/// <param name="skill"></param>
+		/// <param name="caster"></param>
+		/// <param name="originPos"></param>
+		/// <param name="farPos"></param>
+		/// <param name="designatedTarget"></param>
+		public void Handle(Skill skill, ICombatEntity caster, Position originPos, Position farPos, ICombatEntity designatedTarget)
+		{
+			if (!caster.TrySpendSp(skill))
+			{
+				caster.ServerMessage(Localization.Get("Not enough SP."));
+				return;
+			}
+
+			skill.IncreaseOverheat();
+			caster.TurnTowards(designatedTarget);
+			caster.Components.Get<CombatComponent>().SetAttackState(true);
+
+			// Get splash area
+			var splashAreaHeight = 40;
+			var splashAreaWidth = 25;
+
+			// We'll ignore the data sent by the client and get the positions ourselves
+			originPos = caster.Position;
+			farPos = originPos.GetRelative(caster.Direction, splashAreaHeight);
+
+			var splashArea = new Square(originPos, caster.Direction, splashAreaHeight, splashAreaWidth);
+
+			Debug.ShowShape(caster.Map, splashArea, edgePoints: false);
+
+			Send.ZC_SKILL_READY(caster, skill, originPos, farPos);
+
+			this.Attack(skill, caster, splashArea, farPos);
+		}
+
+		/// <summary>
+		/// Executes the actual attack after a delay.
+		/// </summary>
+		/// <param name="skill"></param>
+		/// <param name="caster"></param>
+		/// <param name="splashArea"></param>
+		private async void Attack(Skill skill, ICombatEntity caster, ISplashArea splashArea, Position farPos)
+		{
+			var hits = new List<SkillHitInfo>();
+			
+			//HardCoded Values - @TODO: Make it dynamic
+			var SkillFactor = 0.61f;
+			var AoeAttackRatio = 6;
+
+			var hitTime = TimeSpan.FromMilliseconds(300);
+			var secondHitTime = TimeSpan.FromMilliseconds(550);
+			var damageDelay = TimeSpan.FromMilliseconds(270);
+			var damageSecondDelay = TimeSpan.FromMilliseconds(550);
+			var skillHitDelay = TimeSpan.Zero;
+			
+			var targets = caster.Map.GetAttackableEntitiesIn(caster, splashArea);
+
+			if (targets.Any() && targets.Count > 6)
+			{
+				targets = targets.GetRange(0, AoeAttackRatio);
```

Comment:
After briefly reading up on what exactly the AoE Attack Ratio is supposed to do I must ask, is this actually correct for the current version of the game or is it more of an approximation because we don't have this implemented yet?

The way I understand it, we'd have to factor in the monsters' AoE Attack/Defense Ratios (SDR) and not only stop hitting targets once we reach our own SDR, but also sort them by this value and potentially their distance, right?

Reference: https://www.reddit.com/r/treeofsavior/comments/3w3kjv/aoe_attack_ratio/

Additionally, the skill states it has an SDR of **+6**. That sounds to me like a modifier on top of the character's default SDR. Should the ratio be higher than 6 then?

It's fine if you just added this as a temporary solution, though a comment about it can be helpful for reference in such cases.

Reply:
great!